test/unit/DCAStrategies/dca-strategies-positions-handler.spec.ts

```diff
@@ -445,4 +458,256 @@ contract('DCAStrategiesPositionsHandler', () => {
       });
     });
   });
+
+  describe('syncPositionToNewVersion', () => {
+    let tx: TransactionResponse;
+    let totalAmount = ethers.utils.parseEther('50');
+    let delta = ethers.utils.parseEther('15');
+    let amountOfSwaps = BigNumber.from(5);
+    let newAmountOfSwaps = BigNumber.from(7);
+    let newPositions: BigNumber[];
+    let swapInterval = BigNumber.from(5);
+    let oldShares: IDCAStrategies.ShareOfTokenStruct[];
+    let newShares: IDCAStrategies.ShareOfTokenStruct[];
+
+    when('caller does not have permissions', () => {
+      given(async () => {
+        await DCAStrategiesPositionsHandlerMock.setPermissions(false);
+      });
+      then('tx reverts with message', async () => {
+        await expect(
+          DCAStrategiesPositionsHandlerMock.syncPositionToNewVersion(1, 3, user.address, user.address, amountOfSwaps, newAmountOfSwaps)
+        ).to.be.revertedWith('NoPermissions()');
+      });
+    });
+    when('syncPositionToNewVersion is called', () => {
+      given(async () => {
+        const positions = [1, 2, 3];
+        const OLD_SHARE_A = { token: tokenA.address, share: BigNumber.from(33e2) }; // nothing
```

Comment:
But now we are not testing reduce 😅 

Could we try to test all situations? Maybe adding an extra token?